src/pages/index.js

```diff
@@ -102,11 +179,52 @@ profileEditButton.addEventListener("click", () => {
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit,
-  "#profile-edit-form"
+  settings
 );
 editProfilePopup.setEventListeners();
 
-function handleProfileEditSubmit(inputValuesData) {
-  userInfo.setUserInfo(inputValuesData);
-  editProfilePopup.close();
+function handleProfileEditSubmit(values) {
+  editProfilePopup.renderSaving(true);
+  api
+    .editProfile(values)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfilePopup.renderSaving(false);
+    });
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////-Avatar Modal-///////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+profileEditImage.addEventListener("click", () => profileEditAvatar.open());
+
+const profileEditAvatar = new PopupWithForm(
+  "#avatar-modal",
+  handleProfileAvatar,
+  settings
+);
+profileEditAvatar.setEventListeners();
+
+function handleProfileAvatar(values) {
+  profileEditAvatar.renderSaving(true);
+  api
+    .setAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      profileEditAvatar.close();
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors